HealthcareSystem/GraphicalEditor/MainWindow.xaml.cs

```diff
@@ -129,30 +130,60 @@ private void Edit_Display_Information(object sender, RoutedEventArgs e)
             EditMode = !EditMode;
         }
 
-        private void Change_Display_Information(object sender, RoutedEventArgs e)
-        {          
-            string selectedItem = (string)chooser.SelectedItem;
-            DisplayMapObject.MapObjectType.ObjectTypeFullName = selectedItem;
+        public void ChangeEditButtonVisibility() {
+            if (_selectedMapObject == null)
+            {
+                EditObjectButton.Visibility = Visibility.Hidden;
+            }
+            else
+            {
+                if (_selectedMapObject.MapObjectEntity.MapObjectType.TypeOfMapObject != TypeOfMapObject.ROAD)
+                {
+                    EditObjectButton.Visibility = Visibility.Visible;
+                }
+                else {
+                    EditObjectButton.Visibility = Visibility.Hidden;
+                }
+            }
+        }
+
+
+        private MapObject CreateEditedObject() {
+            MapObjectType type = (MapObjectType)chooser.SelectedItem;
             MapObject objectToEdit = SelectedMapObject;
+            DisplayMapObject.MapObjectType = type;
             objectToEdit.MapObjectEntity = DisplayMapObject;
-            MapObjectType type = new MapObjectType();
-            type.ObjectTypeFullName = DisplayMapObject.MapObjectType.ObjectTypeFullName;           
-            MapObjectEntity newEntity = new MapObjectEntity(type.TypeOfMapObject, DisplayMapObject.Description);
-            MapObject newObject = new MapObject(newEntity, SelectedMapObject.MapObjectMetrics, SelectedMapObject.MapObjectDoor);
-            newObject.MapObjectEntity.Id = SelectedMapObject.MapObjectEntity.Id;
-            _mapObjectController.UpdateMapObject(objectToEdit);                      
+            objectToEdit.Rectangle.Fill = type.ObjectTypeColor;
+            _mapObjectController.UpdateMapObject(objectToEdit);
+            return objectToEdit;
+        }
+
+        private void ReloadMap(MapObject objectToEdit) {
             EditMode = !EditMode;
+            Canvas.Children.Clear();
+            SelectedMapObject.MapObjectEntity.MapObjectType = objectToEdit.MapObjectEntity.MapObjectType;
+            LoadMapOnCanvas();
+        }
+
+        private void Change_Display_Information(object sender, RoutedEventArgs e)
+        {         
+            MapObject mapObjectToEdit = CreateEditedObject();
+            ReloadMap(mapObjectToEdit);
         }
```

Comment:
This is a much better @kmilena98 ! Keep going, one more suggestion is to move these methods:

```sh
  CreateEditedObject();
  ReloadMap(mapObjectToEdit);

```

to separated class, because that is logic outside of MainWindow, and if someone read MainWindow they would only see Change_Display_Information() method and that is enough.

Reply:
@vlaksi We plan to extract as much code from mainWindow as possible to separate classes to refactor mainWindow and avoid it becoming chaotic. This will definitely be done in some of next commits.

Reply:
That is great @VasilijeBursac